sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -110,23 +155,165 @@ protected string GetSessionFilePath()
         /// This will run once before any tests.
         /// </summary>
         [OneTimeSetUp]
-        public void StartLoggingEvents()
+        public void InitializeRecordedTestClass()
         {
+            TestContext.Progress.WriteLine($"starting up {GetType().FullName}");
+
             if (Mode == RecordedTestMode.Live || Debugger.IsAttached)
             {
                 Logger = new TestLogger();
             }
+
+            if (!_useLegacyTransport)
+            {
+                StartTestProxy();
+            }
+        }
+
+        private void StartTestProxy()
+        {
+            var testProxyPath = GetType().Assembly.GetCustomAttributes<AssemblyMetadataAttribute>().Single(a => a.Key == "TestProxyPath")
+                .Value;
+            TestContext.Progress.WriteLine(testProxyPath);
+            ProcessStartInfo testProxyProcessInfo = new ProcessStartInfo(
+                s_dotNetExe,
+                testProxyPath)
+            {
+                UseShellExecute = false,
+                EnvironmentVariables =
+                    { { "ASPNETCORE_URLS", "http://0.0.0.0:0;https://0.0.0.0:0" }, { "Logging__LogLevel__Microsoft", "Information" } },
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+
+            // Due to inherent race condition involving cleaning and importing the dev cert in different RecordedTestBase instances (which may be in different processes),
+            // attempt to start up to three times.
+            for (int attempt = 1; attempt <= 3; attempt++)
+            {
+                if (!TestEnvironment.GlobalIsRunningInCI)
+                {
+                    ImportDevCertIfNeeded();
+                }
+
+                _proxyPortHttp = null;
+                _proxyPortHttps = null;
+
+                _testProxyProcess = Process.Start(testProxyProcessInfo);
```

Comment:
Copy the JobObject management code from autorest and use it so we don't leak proxy processes.